homeassistant/components/thermopro/button.py

```diff
@@ -0,0 +1,98 @@
+"""Demo platform that offers a fake button entity."""
+
+from __future__ import annotations
+
+import logging
+
+from thermopro_ble import ThermoProDevice
+
+from homeassistant.components import persistent_notification
+from homeassistant.components.bluetooth import async_ble_device_from_address
+from homeassistant.components.button import ButtonEntity
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import device_registry as dr
+from homeassistant.helpers.device_registry import DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.util.dt import now
+
+from . import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the demo button platform."""
+
+    assert config_entry.unique_id is not None
+
+    device_registry = dr.async_get(hass)
+    device = device_registry.async_get_device(
+        connections={(dr.CONNECTION_BLUETOOTH, config_entry.unique_id)},
+        identifiers={(dr.CONNECTION_BLUETOOTH, config_entry.unique_id)},
+    )
+
+    assert device is not None
+
+    _LOGGER.debug("UHUHUHUH %s - %s - %s", config_entry, config_entry.data, device)
+
+    if device.model not in ("TP358", "TP393"):
+        return
+
+    async_add_entities(
+        [
+            ThermoProButton(
+                address=config_entry.unique_id,
+                unique_id="push",
+                device_name="Push",
+            ),
+        ]
+    )
+
+
+class ThermoProButton(ButtonEntity):
+    """Representation of a demo button entity."""
+
+    _attr_has_entity_name = True
+    _attr_name = None
+    _attr_should_poll = False
+
+    def __init__(
+        self,
+        address: str,
+        unique_id: str,
+        device_name: str,
+    ) -> None:
+        """Initialize the Demo button entity."""
+        self.address = address
+        self._attr_unique_id = unique_id
+        self._attr_device_info = DeviceInfo(
+            identifiers={(DOMAIN, unique_id)},
+            name=device_name,
+        )
```

Comment:
Match the logic to what bthome does here https://github.com/home-assistant/core/blob/c965dfb619f99fe3e2a8832b4f7cffb1c71ec56e/homeassistant/components/bthome/event.py#L73 and it will link to the existing device